anna/my-app/src/pages/home/HomePage.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import Layout from '../../components/layout';
+import './homePage.css';
+import MovieList from '../../components/movieList';
+import IMovie from '../../components/movieList/movie-card/IMovie';
+import SearchForm from '../../components/searchForm';
+import { ISearchFormState } from '../../components/searchForm/SearchForm';
+import SortBox from '../../components/sortBox';
+import Container from '../../components/container';
+
+import * as QueryString from "query-string";
+import { RouteComponentProps } from "react-router-dom";
+import Api from '../../Api';
+
+interface IHomeProps {
+  movies: IMovie[];
+  changeMovies: (movies: IMovie[]) => void;
+}
+
+interface IHomeState {
+  sortBy: string;
+}
+
+class HomePage extends React.Component<IHomeProps & RouteComponentProps, IHomeState> {
+
+  state: IHomeState = {
+    sortBy: "release_date"
+  }
+
+  componentDidMount() {
+    this.changeMoviesByPath();
+  }
+
+  componentDidUpdate(prevProps: RouteComponentProps) {
+    const { location } = this.props;
+    if (prevProps.location.search !== location.search) {
+      this.changeMoviesByPath();
+    }
+  }
+
+  changeMoviesByPath = async(): Promise<void> => {
+    const { changeMovies, location } = this.props;
+    
+    if (location.pathname === "/") {
+      const movies = await Api.fetchMovies('?sortBy=release_date');
+      changeMovies(movies);         
+        
+    } else if (location.pathname === "/search") {
+      const movies = await Api.fetchMovies(location.search)
+      changeMovies(movies);      
+    }
+  }
```

Comment:
Pass object instead of string. You can add default values, so you won't need this check